tensorflow_datasets/text/tasks.py

```diff
@@ -0,0 +1,162 @@
+related to the dataset and with the id number related to the answer
+"""
+
+_DOWNLOAD_URL = ("http://www.thespermwhale.com/jaseweston/babi/"
+                 "tasks_1-20_v1-2.tar.gz")
+_TOP_LEVEL_DIR = "tasks_1-29_v1-2"
+
+_TRAIN_FILE_FORMAT = os.path.join(_TOP_LEVEL_DIR, "shuffled-10k", "qa(.*)train.txt")
+_TEST_FILE_FORMAT = os.path.join(_TOP_LEVEL_DIR, "shuffled-10k", "qa(.*)test.txt")
+
+class TasksConfig(tfds.core.BuilderConfig):
+    """ Builder config for tasks dataset"""
+    @api_utils.disallow_positional_args
+    def __init__(self, text_encoder_config=None, **kwargs):
+        """ BuilderConfig for tasks.
+        Args:
+      text_encoder_config: `tfds.features.text.TextEncoderConfig`, configuration
+        for the `tfds.features.text.TextEncoder` used for the tasks `"text"`
+        feature.
+      **kwargs: keyword arguments forwarded to super.
+    """
+        super(TasksConfig, self).__init__(**kwargs)
+        self.text_encoder_config = (
+            text_encoder_config or tfds.features.text.TextEncoderConfig())
+
+def _train_data_filenames(tmp_dir):
+    return tf.io.gfile.glob(os.path.join(tmp_dir, _TRAIN_FILE_FORMAT))
+
+def _test_data_filenames(tmp_dir):
+    return tf.io.gfile.glob(os.path.join(tmp_dir, _TEST_FILE_FORMAT))
+
+class tasks(tfds.core.GeneratorBasedBuilder):
+    """ Tasks dataset combining useful information of training and test data """
+    BUILDER_CONFIGS = [
+        TasksConfig(
+            name="plain_text",
+            version="0.0.1",
+            description="Plain text",
+        ),
+        TasksConfig(
+            name="bytes",
+            version="0.0.1",
+            description=("Uses byte-level text encoding with "
+                         "`tfds.features.text.ByteTextEncoder`"),
+            text_encoder_config=tfds.features.text.TextEncoderConfig(
+                encoder=tfds.features.text.ByteTextEncoder()),
+        ),
+        TasksConfig(
+            name="subwords8k",
+            version="0.0.2",
+            description=("Uses `tfds.features.text.SubwordTextEncoder` with 8k "
+                         "vocab size"),
+            text_encoder_config=tfds.features.text.TextEncoderConfig(
+                encoder_cls=tfds.features.text.SubwordTextEncoder,
+                vocab_size=2**13),
+        ),
+        TasksConfig(
+            name="subwords32k",
+            version="0.0.2",
+            description=("Uses `tfds.features.text.SubwordTextEncoder` with "
+                         "32k vocab size"),
+            text_encoder_config=tfds.features.text.TextEncoderConfig(
+                encoder_cls=tfds.features.text.SubwordTextEncoder,
+                vocab_size=2**15),
+        ),
+    ]
+
+
+    def _info(self):
+        # Specifies tfds.core.DatasetInfo object
+        return tfds.core.DatasetInfo(
+            builder=self,
+            description=_DESCRIPTION,
+            features=tfds.features.FeaturesDict({
+                "text":tfds.features.Text(
```

Comment:
This dataset has a lot more context and representing it 'only as text' - is losing most of it.

the features should be as SequenceDict, and inside it should have:
- id, text and (optionally): answer + reference